### PR TITLE
Fix #1734: Limit the amount of history and bookmarks shown.

### LINKED ISSUE #1734: [#868 Followup] Restrict number of suggestions from history/bookmark
### Description: 
[#868 Followup] Restrict number of suggestions from history/bookmark

### Steps to Reproduce 
  1. Visit a bunch of sites
  2. Add a few of them as bookmarks
  3. Open a new tab and type in a letter, shows a whole list of suggestions from history/bookmarks

**Actual result:** 
image

**Expected result:**
As per https://github.com/brave/brave-ios/issues/868#issuecomment-543324310 it should be restricted to 5

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.13

**Device details:** 
All

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary of Changes

This pull request fixes issue https://github.com/brave/brave-ios/issues/1734

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Let's clamp the size where it's assigned rather than just on the number of rows: https://github.com/brave/brave-ios/blob/faf7685088e2103c040b513b2f41916a1af94cbc/Client/Frontend/Browser/Search/SearchViewController.swift#L394

**Review by iccub — CHANGES_REQUESTED** [review]
The best would be to limit it at query level, no?

**Comment by Brandon-T** [comment]
> The best would be to limit it at query level, no?

I would have to change it in `_SearchLoader` class which might be used elsewhere as well.

**Review by iccub — APPROVED** [review]

File: Client/Frontend/Browser/Search/SearchViewController.swift [hunk 1]
```diff
@@ -389,7 +389,7 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
     }
 
     func loader(dataLoaded data: [Site]) {
-        self.data = data
+        self.data = data.count == 5 ? data : Array(data.dropFirst(5))
```
Comment (kylehickinson) [inline Client/Frontend/Browser/Search/SearchViewController.swift]: I don't understand, why `dropFirst`? Also why checking == 5?

If this comes back with 7 results, you're dropping the first 5 results–resulting in 2 left–not limiting the overall list size to 5

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: Client/Frontend/Browser/Search/SearchViewController.swift (hunk 1 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Search/SearchViewController.swift]: @kylehickinson Fixed it. I was reading the documentation wrong. I changed it to an array slice.

File: Client/Frontend/Browser/Search/SearchViewController.swift [hunk 2]
```diff
@@ -389,7 +389,7 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
     }
 
     func loader(dataLoaded data: [Site]) {
-        self.data = data
+        self.data = data.count <= 5 ? data : Array(data[0..<5])
```
Comment (kylehickinson) [inline Client/Frontend/Browser/Search/SearchViewController.swift]: Lets replace this with `data.prefix(5)` 🙂

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**